### DISCUSSION TITLE
Support for private images?

### DISCUSSION THREAD

**Discussion by K8sKween** [nicklambourne/slackblocks#83] [Ideas]
Slack announcement here: [slack.com/blog/developers/uploading-private-images-blockkit](https://slack.com/blog/developers/uploading-private-images-blockkit)

One of the places that this feature might divert from the existing use of this library, that being if we had a abstraction for the private image as a file object it would involve this library not just building json blocks but create new dependencies.

It might make more sense to just update the block objects to accept the file_id and the user would be responsible for uploading it with another SDK. 

Thoughts?

**Comment by nicklambourne** [comment]
Apologies for the delayed response. I can see valid arguments for going either way.

It should be possible to use the Python standard library (I'm guessing urllib3, instead of something like requests) to accomplish the former, or if not, use the Python "extras" packaging functionality to allow people to opt in to that functionality and the extra dependencies it would require.

The latter would be more in line with the core mission of the library (and my current time constraints), but I'd be tempted to implement this as a first step, even if there was going to be a later attempt to provide that extended functionality.
